core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -95,10 +148,18 @@ private[spark] object ExecutorMetricType {
     OnHeapUnifiedMemory,
     OffHeapUnifiedMemory,
     DirectPoolMemory,
-    MappedPoolMemory
+    MappedPoolMemory,
+    ProcessTreeMetrics
   )
 
-  // Map of executor metric type to its index in values.
-  val metricIdxMap =
-    Map[ExecutorMetricType, Int](ExecutorMetricType.values.zipWithIndex: _*)
+  var definedMetricsAndOffset = mutable.LinkedHashMap.empty[String, Int]
+  var numberOfMetrics = 0
```

Comment:
you only need this mutable during initialization, so to convey its eventually final value, you can move the mutability to inside a block:

```scala
val (metricToOffset, numMetrics) = {
  var n = 0
  val _metricToOffset = mutable.LinkedHashMap.empty[String, Int]
  ...
   (_metricToOffset, n)
}
```